code/datums/weather/weather.dm

```diff
@@ -426,6 +426,8 @@
 		hit_mob.electrocute_act(50, "thunder", flags = SHOCK_TESLA|SHOCK_NOGLOVES)
 
 	for(var/obj/hit_thing in weather_turf)
+		if(QDELETED(hit_thing)) // stop, it's already dead
+			continue
 		hit_thing.take_damage(20, BURN, ENERGY, FALSE)
```

Comment:
This should also check for uses_integrity, invisibility and TRAIT_UNDERFLOOR while you're at it. ``/obj/effect`` doesn't use integrity, and invisible objects are most likely abstract holders